news-explorer-frontend/src/components/App/App.js

```diff
@@ -0,0 +1,394 @@
+            name: res[0].name,
+            id: res[0]._id,
+          });
+        })
+        .catch((error) => console.log(error));
+      api.getInitialArticles()
+        .then((res) => {
+          // console.log('test', res);
+          setSavedArticles(res);
+        });
+    }
+  }, [loggedIn]);
+
+  const mobileNavStyleHandler = () => {
+    if (isMobileNavOpen && path === '/') {
+      setNavBackgroundTypeClass('nav-container_theme_dark');
+      setMobileLinkTypeClass('mobile-navigation__link_type_bright');
+      setNavMobileBtnClass('navigation__mobile-btn_close_bright');
+      setHeaderDarkOverlay('header_dark-overlay');
+    }
+
+    if (isMobileNavOpen && path === '/saved-news') {
+      setNavBackgroundTypeClass('nav-container_theme_bright');
+      setMobileLinkTypeClass('mobile-navigation__link_type_dark');
+      setNavMobileBtnClass('navigation__mobile-btn_close_dark');
+    }
+  };
+
+  const navStyleHandler = () => {
+    if (path === '/') {
+      setCurrentTypeClass('navigation__list-item_bright-border');
+      setLinkColorClass('');
+      setNavBackgroundTypeClass('');
+      setCurrentPage('home');
+      setMobileNavTypeClass('mobile-navigation_theme_dark');
+      setNavMobileBtnClass('navigation__mobile-btn_menu_bright');
+      setHeaderDarkOverlay('');
+
+      mobileNavStyleHandler();
+    }
+    if (path === '/saved-news') {
+      setCurrentTypeClass('navigation__list-item_dark-border');
+      setLinkColorClass('navigation__link_type_dark');
+      setNavBackgroundTypeClass('navigation_theme_bright');
+      setCurrentPage('saved articles');
+      setMobileNavTypeClass('mobile-navigation_theme_bright');
+      setNavMobileBtnClass('navigation__mobile-btn_menu_dark');
+
+      mobileNavStyleHandler();
+    }
+  };
+
+  useEffect(() => {
+    navStyleHandler();
+  }, [isMobileNavOpen]);
+
+  const closePopup = () => {
+    setIsSignInPopupOpen(false);
+    setIsInfoTooltipOpen(false);
+    setIsRegisterPopupOpen(false);
+  };
+
+  const handleLoggedIn = () => {
+    setLoggedIn((prevState) => !prevState);
+  };
+
+  const handleSearchCards = (keyword) => {
+    setShowPreloaderClass('preloader_visible');
+    getArticles(keyword)
+      .then((res) => {
+        setArticles(res.articles);
+        if (res.articles.length) {
+          setFoundArticles(true);
+          setShowPreloaderClass('');
+          setIsSearchingCards(false);
+        } else {
+          setFoundArticles(false);
+        }
+        setKeyword(keyword);
+      })
+      .catch((err) => {
+        setFoundArticles(false);
+        console.log('err', err);
+      });
+  };
+
+  const SaveArticleHandler = (data) => {
+    if (loggedIn) {
+      api.saveArticle(data)
+        .then((article) => {
+          const newArticle = { title: data.title, ...article };
+          setSavedArticles([newArticle, ...savedArticles]);
+        })
+        .catch((err) => console.log(err));
+    }
+  };
+
+  const articleDeleteHandler = (article) => {
```

Comment:
Function names should start with a verb in infinitive (there is such a point in the check list). It’s common in coding to do it so that functions could `speak` for themselves about what they do.


`deleteArticle`